controllers/posts.controller.js

```diff
@@ -0,0 +1,162 @@
+    });
+    return;
+  }
+  
+  const {user} = req.user;
+  id = user.id;
+
+  const post = {
+    title: req.body.title,
+    content: req.body.content,
+    published: req.body.published,
+    userId: id,
+  }
+
+  posts.create(post)
+    .then((post) => {
+      res.send(post);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: err.message || "Some error occurred"
+      });
+    });
+};
+
+exports.findPostsByUserId = (req, res) => {
+
+  const {user} = req.user;
+  id = user.id;
+
+  posts.findAll({ where: { userId: id } })
+    .then((post) => {
+      if (post) {
+        res.send(post);
+      }
+      else {
+        res.status(404).send({
+          message: 'Cannot find any post with UserId=' + id
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: "Error retrieving posts with Userid=" + id
+      });
+    });
+};
+
+exports.findPublishedPostsByUserId = (req, res) => {
+
+  const {user} = req.user;
+  id = user.id;
+
+  posts.findAll({ where: { userId: id, published: true } })
+    .then((post) => {
+      if (post) {
+        res.send(post);
+      }
+      else {
+        res.status(404).send({
+          message: 'Cannot find any published post by User No: ' + id
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: "Error retrieving posts with Userid=" + id
+      });
+    });
+};
+
+exports.findAllPublishedPosts = (req, res) => {
+
+  posts.findAll({ where: { published: true } })
+    .then((posts) => {
+      if (posts) {
+        res.send(posts);
+      }
+      else {
+        res.status(404).send({
+          message: 'Cannot find any published post'
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: "Error retrieving posts"
+      });
+    });
+};
+
+
+exports.findDraftedPostsByUserId = (req, res) => {
+
+  const {user} = req.user;
+  id = user.id;
+
+  posts.findAll({ where: { userId: id, published: false } })
+    .then((post) => {
+      if (post) {
+        res.send(post);
+      }
+      else {
+        res.status(404).send({
+          message: 'Cannot find any drafted post with by User No: ' + id
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: "Error retrieving posts with Userid=" + id
+      });
+    });
+};
+
+
+exports.DeletePost = (req, res) => {
+  const pid = req.body.postId;
+  
+  posts.destroy({ where: { id: pid } })
+    .then((post) => {
+      if (post == 1) {
+        res.send("Post deleted successfully");
+      }
+      else {
+        res.status(404).send({
+          message: 'Cannot find any post with id= ' + pid
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: "Error retrieving posts with id= " + pid
+      });
+    });
+};
+
+exports.UpdatePost = (req, res) => {
+  const pid = req.body.postId;
```

Comment:
post id should be read from query params. First need to make sure if the current user is owner of the post else he return 401 error